src/Engine/Physics/Shape.hpp

```diff
@@ -67,9 +67,10 @@ namespace Physics {
             /**
              * @return The Bullet shape.
              */
-            btCollisionShape* GetShape();
+            btCollisionShape* GetShape() const;
+
+            Shape(const Shape& other) = delete;
 
-        private:
             btCollisionShape* shape;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Class 'Shape' is unsafe, 'Shape::shape' can leak by wrong usage. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunsafeClassCanLeak)